### PR TITLE
docs/flow: Document local.file and secrets

### PR DESCRIPTION
**Pull request by rfratto**
This PR creates documentation for the newly introduced `local.file` component (#1746). 

I've temporarily placed this in `docs/flow` while Grafana Agent Flow is still being worked on. This should eventually be moved to the `docs/sources` once the MVP is ready for shipping to users.

Note that there is much more to document regarding Grafana Agent Flow, and this is just serves as an example for how reference-level documentation for components should be written. Higher level documentation explaining how to use Flow and common use cases should be done in a later PR.

### CODE REVIEW

File: docs/flow/components/local.file.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+The `poll` file change detector will cause the watched file to be reread
+every `poll_frequency`, regardless of whether the file changed.
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`content` | `string` or `secret` | The contents of the file from the most recent read
```
Comment (mattdurham) [inline docs/flow/components/local.file.md:67]: should we use the term sensitive here too?

Comment (rfratto) [reply docs/flow/components/local.file.md:67]: The type is called `secret` internally, maybe we should use `secret` consistently instead of sensitive?

Comment (mattdurham) [reply docs/flow/components/local.file.md:67]: Sounds good

**Review by mattdurham — APPROVED** [review]
Lgtm

**Review by brendamuir — APPROVED** [review]

**Review by tpaschalis — APPROVED** [review]

**Pull request merged**